superset-frontend/src/views/App.tsx

```diff
@@ -68,30 +69,70 @@ const LocationPathnameLogger = () => {
   return <></>;
 };
 
-const App = () => (
-  <Router>
-    <ScrollToTop />
-    <LocationPathnameLogger />
-    <RootContextProviders>
-      <GlobalStyles />
-      <Menu
-        data={bootstrapData.common.menu_data}
-        isFrontendRoute={isFrontendRoute}
-      />
-      <Switch>
-        {routes.map(({ path, Component, props = {}, Fallback = Loading }) => (
-          <Route path={path} key={path}>
-            <Suspense fallback={<Fallback />}>
-              <ErrorBoundary>
-                <Component user={bootstrapData.user} {...props} />
-              </ErrorBoundary>
-            </Suspense>
-          </Route>
-        ))}
-      </Switch>
-      <ToastContainer />
-    </RootContextProviders>
-  </Router>
-);
+function hasOnlyGuestRole(data: BootstrapData) {
+  // Check if we have user data and roles
+  if (!data?.user?.roles) {
+    return false;
+  }
+
+  const { roles } = data.user;
+
+  // Get all role keys
+  const roleKeys = Object.keys(roles);
+
+  // Check if there's exactly one role
+  if (roleKeys.length !== 1) {
+    return false;
+  }
+
+  // Check if that single role is "Guest"
+  return roleKeys[0] === 'Guest';
+}
```

Comment:
This can be simplified to return ` !(roleKeys.length !== 1) && roleKeys[0] === 'Guest';`